memfs/memory.go

```diff
@@ -375,3 +391,16 @@ func isReadOnly(flag int) bool {
 func isWriteOnly(flag int) bool {
 	return flag&os.O_WRONLY != 0
 }
+
+func isInDir(dir, path string) bool {
+	p, err := filepath.Rel(dir, path)
+	if err != nil {
+		return false
+	}
+
+	if p == "." || p == ".." {
+		return false
```

Comment:
well, I haven't found any case where this can be reproduced from public methods...